packages/contracts/src/manager/facets/ManagerFacet.sol

```diff
@@ -0,0 +1,251 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.3;
+
+import {Modifiers} from "../libraries/LibAppStorage.sol";
+import {AccessControlStorage} from "../libraries/AccessControlStorage.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../../dollar/interfaces/IUbiquityDollarToken.sol";
+import "../../dollar/interfaces/ICurveFactory.sol";
+import "../../dollar/interfaces/IMetaPool.sol";
+import "../../dollar/core/TWAPOracleDollar3pool.sol";
+
+contract ManagerFacet is Modifiers {
+    // TODO Add a generic setter for extra addresses that needs to be linked
+    function setTwapOracleAddress(address _twapOracleAddress)
+        external
+        onlyAdmin
+    {
+        s.twapOracleAddress = _twapOracleAddress;
+        // to be removed
+
+        TWAPOracleDollar3pool oracle =
+            TWAPOracleDollar3pool(s.twapOracleAddress);
+        oracle.update();
+    }
+
+    function setDollarTokenAddress(address _dollarTokenAddress)
```

Comment:
In the diamond Init it is pat of the diamond so we don't need it here as it will always be the diamond address

Reply:
> In the diamond Init it is pat of the diamond so we don't need it here as it will always be the diamond address

Explained above